### PR TITLE
DeepChem Feedstock 2.4.0 Update

### PR DESCRIPTION
**Pull request by rbharath**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by rbharath** [comment]
@simonbray @nd-02110114 Could you take a quick look at this update to see it it looks sensible? This is my first time personally updating the feedstock so still figuring it all out :)

**Comment by rbharath** [comment]
@conda-forge-admin, please rerender

**Comment by nissy-dev** [comment]
LGTM !

However, the CI failed when building. The error message said the conflict between incompatible packages. Considering error message, I seem the pip version may not be good. The new pip (>=20.3.3) resolve the dependencies strictly, so sometimes we face installation error. I faced the similar error when migrating GitHub Actions from Travis.

**Comment by simonbray** [comment]
It seems TF version `2.*` are not available on conda-forge, which is a real shame. :( See https://github.com/conda-forge/tensorflow-feedstock/issues/107.

However, you are only using it for testing. Can you run the test with an older version instead?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
-  script: "PBR_VERSION=${PKG_VERSION} && export PBR_VERSION && {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+  number: 0
+  script: PBR_VERSION=${PKG_VERSION} && export PBR_VERSION && {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
 
 requirements:
   host:
-    - python >=3.5
-    - pip
+    - python >=3.5,<3.8
+    - pip==20.2.*
```
Comment (simonbray) [inline recipe/meta.yaml]:
```suggestion
    - pip
```
pip is not used for resolving dependencies, just for the installation of deepchem (see the script line above in the build section) so pinning here doesn't solve the problem

**Comment by nissy-dev** [comment]
> Can you run the test with an older version instead

Unfortunately, DeepChem 2.4.0 doesn't work with `TF 1.*` 
I seem it is not necessary to run the test script and we can remove the old test script.

**Comment by rbharath** [comment]
Thanks for the feedback! I've commented out the TF 2.x test and removed the unnecessary pip pin here. 

@simonbray @nd-02110114 Can you take a quick second look? The CI is green so I think we can merge this in now

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-    - python >=3.5
+    - python >=3.5,<3.8
     - pip
     - setuptools
     - pbr
 
   run:
-    - python >=3.5
+    - python >=3.5,<3.8
+    - pip
```
Comment (simonbray) [inline recipe/meta.yaml]: pip is not needed at runtime, just for installation, so it can be removed here

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
     - joblib
     - numpy
     - pandas
-    - pillow # pillow will remove from v2.4.0
-    - scikit-learn <=0.22.1
+    - scikit-learn
     - scipy
 
 test:
   requires:
```
Comment (simonbray) [inline recipe/meta.yaml:33]: if the tests are being removed, this `requires` section can also be removed / commented out, or at least tensorflow

File: recipe/meta.yaml (hunk 2 above)
Comment (rbharath) [reply recipe/meta.yaml]: Good point! I've removed TF but kept in rdkit since one of the tests does a rdkit sanity import check

**Review by simonbray — APPROVED** [review]
Looks good to me!

**Pull request merged**